### PR TITLE
Adding NCrystal

### PR DESCRIPTION
**Pull request by tkittel**
This is a PR to add a conda-forge recipe for the open source project NCrystal (https://github.com/mctools/ncrystal).

I did review the checklist and have also successfully tested the recipe locally on my linux ubuntu 20.04 machine.

Checklist

- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see here for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (url) rather than a repo (e.g. git_url) is used in your recipe (see here for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our knowledge base documentation before pinging a team.

### CODE REVIEW

**Comment by tkittel** [comment]
@conda-forge/staged-recipes ready for review!

**Comment by tkittel** [comment]
@conda-forge/help-c-cpp The PR is ready for review.

File: recipes/ncrystal/build.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+set -x
+set -e
+which cmake
+cmake --version
+$PYTHON --version
```
Comment (ocefpaf) [inline recipes/ncrystal/build.sh:6]: These are well defined in conda-forge and logged. I'm saying you have to remove but you probably don't need them here.

File: recipes/ncrystal/build.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    -DBUILD_EXAMPLES=OFF \
+    -DINSTALL_SETUPSH=OFF \
+    -DEMBED_DATA=ON \
+    -DINSTALL_DATA=OFF \
+    -DNO_DIRECT_PYMODINST=ON \
+    -DPython3_EXECUTABLE="$PYTHON" \
+    ${CMAKE_ARGS}
+
+make -j${CPU_COUNT:-1}
+make install
```
Comment (ocefpaf) [inline recipes/ncrystal/build.sh:28]: Is there a `make check` or `ctest` we can run here?

File: recipes/ncrystal/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
+{% set org = "mctools" %}
+{% set name = "ncrystal" %}
+{% set version = "3.3.0" %}
+{% set sha256 = "0a1e1c818e78bf61a0acf9b941fde8b813d74d1332c399d38198789b5c502219" %}
```
Comment (ocefpaf) [inline recipes/ncrystal/meta.yaml:4]:
```suggestion
```

File: recipes/ncrystal/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/{{ org }}/{{ name }}/archive/v{{ version }}.tar.gz
+  fn: {{ name }}-{{ version }}.tar.gz
+  folder: src
+  sha256: {{ sha256 }}
```
Comment (ocefpaf) [inline recipes/ncrystal/meta.yaml:15]:
```suggestion
  sha256: 0a1e1c818e78bf61a0acf9b941fde8b813d74d1332c399d38198789b5c502219
```

File: recipes/ncrystal/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,76 @@
+{% set org = "mctools" %}
+{% set name = "ncrystal" %}
+{% set version = "3.3.0" %}
+{% set sha256 = "0a1e1c818e78bf61a0acf9b941fde8b813d74d1332c399d38198789b5c502219" %}
+{% set build = 0 %}
```
Comment (ocefpaf) [inline recipes/ncrystal/meta.yaml:5]:
```suggestion
```

File: recipes/ncrystal/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/{{ org }}/{{ name }}/archive/v{{ version }}.tar.gz
+  fn: {{ name }}-{{ version }}.tar.gz
+  folder: src
+  sha256: {{ sha256 }}
+
+build:
+  number: {{ build }}
```
Comment (ocefpaf) [inline recipes/ncrystal/meta.yaml:18]:
```suggestion
  number: 0
```

File: recipes/ncrystal/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+  summary: Library for thermal neutron transport in crystals and other materials.
+  description: |
+    NCrystal is a library and toolkit for thermal neutron transport in crystals
+    and other materials. It can be used standalone or as a backend for
+    other Monte Carlo particle simulation frameworks.
+  license: Apache-2.0
+  license_family: Apache
+  license_file: src/LICENSE
+  doc_url: https://github.com/{{ org }}/{{ name|lower }}/wiki
+  dev_url: https://github.com/{{ org }}/{{ name|lower }}
```
Comment (ocefpaf) [inline recipes/ncrystal/meta.yaml:72]: The use of jinja here obfuscates the recipe and make it hard to follow the links when reading the recipe. Please use the full URLs instead.

**Comment by tkittel** [comment]
Thanks a lot for reviewing this @ocefpaf ! I have tried to implement all of your changes, with the exception that I could't add a "make ctest" line, since NCrystal does not provide this feature (but there are tests in the meta.yaml). I have instead added a comment in the build.sh script as a reminder to add such a "make ctest" line in the future if it becomes available.

I also removed all the unnecessary Jinja variable usage as you requested. I was merely trying to mimick the way it was done in some other feedstock repo I was looking at for inspiration, I don't actually particularly need them :smile:

File: recipes/ncrystal/build.sh (hunk 2 above)
Comment (tkittel) [reply recipes/ncrystal/build.sh:26]: There is not, but I have added a comment in the text concerning potentially adding one in the future.
